src/kbmod/reprojection.py

```diff
@@ -0,0 +1,135 @@
+    )
+
+    return new_image, footprint
+
+
+def reproject_work_unit(work_unit, common_wcs):
+    """Given a WorkUnit and a WCS, reproject all of the images in the ImageStack
+    into a common WCS.
+
+    Attributes
+    ----------
+    work_unit : `kbmod.WorkUnit`
+        The WorkUnit to be reprojected.
+    common_wcs : `astropy.wcs.WCS`
+        The WCS to reproject all the images into.
+
+    Returns
+    ----------
+    A `kbmod.WorkUnit` reprojected with a common `astropy.wcs.WCS`.
+    """
+    height, width = common_wcs.array_shape
+    images = work_unit.im_stack.get_images()
+    obstimes = np.array(work_unit.get_all_obstimes())
+
+    if len(work_unit.per_image_wcs) != len(images):
+        raise ValueError("per_image_wcs not provided for all WorkUnit")
+
+    image_list = []
+
+    unique_obstimes = np.unique(obstimes)
+
+    for time in unique_obstimes:
+        indices = list(np.where(obstimes == time)[0])
+
+        science_add = np.zeros(common_wcs.array_shape)
+        variance_add = np.zeros(common_wcs.array_shape)
+        mask_add = np.zeros(common_wcs.array_shape)
+        footprint_add = np.zeros(common_wcs.array_shape)
+
+        for index in indices:
+            image = images[index]
+            science = image.get_science()
+            variance = image.get_variance()
+            mask = image.get_mask()
+            original_wcs = work_unit.per_image_wcs[index]
+
+            reprojected_science, footprint = reproject_raw_image(science, original_wcs, common_wcs, time)
+
+            footprint_add += footprint
+            # we'll enforce that there be no overlapping images at the same time,
+            # for now. We might be able to add some ability co-add in the future.
+            if np.any(footprint_add > 1.0):
+                raise ValueError("Images with the same obstime are overlapping.")
+
+            reprojected_variance, _ = reproject_raw_image(variance, original_wcs, common_wcs, time)
+
+            reprojected_mask, _ = reproject_raw_image(mask, original_wcs, common_wcs, time)
+
+            # change all the NaNs to zeroes so that the matrix addition works properly.
+            # `footprint_add` will maintain the information about what areas of the frame
+            # don't have any data so that we can change it back after we combine.
+            reprojected_science[np.isnan(reprojected_science)] = 0.0
+            reprojected_variance[np.isnan(reprojected_variance)] = 0.0
+            reprojected_mask[np.isnan(reprojected_mask)] = 0.0
+
+            science_add += reprojected_science
+            variance_add += reprojected_variance
+            mask_add += reprojected_mask
+
+        # change all the values where there are is no corresponding data to `KB_NO_DATA`.
+        gaps = footprint_add == 0.0
+        science_add[gaps] = KB_NO_DATA
+        variance_add[gaps] = KB_NO_DATA
+        mask_add[gaps] = 1
+
+        mask_add = np.where(np.isclose(mask_add, 0.0, atol=1e-01), 0.0, 1.0)
```

Comment:
It is worth adding a short comment here that this is effectively transforming the mask back into a 0/1 bit mask.